src/go/k8s/pkg/resources/certmanager/ca_certificate_bundle_test.go

```diff
@@ -0,0 +1,267 @@
+
+	pandaCluster := &v1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterName,
+		},
+		Spec: v1alpha1.ClusterSpec{
+			Replicas:      pointer.Int32(1),
+			Configuration: v1alpha1.RedpandaConfig{},
+		},
+	}
+
+	caCert, err := getCertSecretFromFile("./testdata/ca-cert-secret.yaml")
+	assert.NoError(t, err)
+	caCert.SetNamespace(nsPrefix)
+	err = fakeK8sClient.Create(context.TODO(), caCert)
+	assert.NoError(t, err)
+
+	tlsCert, err := getCertSecretFromFile("./testdata/tls-cert-secret.yaml")
+	assert.NoError(t, err)
+	tlsCert.SetNamespace(nsPrefix)
+	err = fakeK8sClient.Create(context.TODO(), tlsCert)
+	assert.NoError(t, err)
+
+	expectedOneCACertData := getExpectedCACert("./testdata/expected-one-ca-crt.pem")
+	expectedTwoCACertData := getExpectedCACert("./testdata/expected-two-ca-crt.pem")
+	expectedThreeCACertData := getExpectedCACert("./testdata/expected-three-ca-crt.pem")
+
+	table := []struct {
+		scenario         string
+		certs            []*types.NamespacedName
+		expectedBundleCA []byte
+	}{
+		{
+			scenario:         "Get CA cert from secret having ca.crt",
+			certs:            []*types.NamespacedName{{Name: caCert.Name, Namespace: caCert.Namespace}},
+			expectedBundleCA: expectedOneCACertData,
+		},
+		{
+			scenario:         "Get CA certs from secret having tls.crt",
+			certs:            []*types.NamespacedName{{Name: tlsCert.Name, Namespace: tlsCert.Namespace}},
+			expectedBundleCA: expectedTwoCACertData,
+		},
+		{
+			scenario: "Get CA certs from secret having ca.crt and secret having tls.crt",
+			certs: []*types.NamespacedName{
+				{Name: caCert.Name, Namespace: caCert.Namespace},
+				{Name: tlsCert.Name, Namespace: tlsCert.Namespace},
+			},
+			expectedBundleCA: expectedThreeCACertData,
+		},
+	}
+	for i, tt := range table {
+		t.Run(fmt.Sprintf("case-%d: %s", i, tt.scenario), func(t *testing.T) {
+			ns := fmt.Sprintf("%s-%d", nsPrefix, i)
+			pandaCluster.SetNamespace(ns)
+
+			caCertBundle := certmanager.NewCACertificateBundle(fakeK8sClient, fakeK8sClient.Scheme(), pandaCluster,
+				tt.certs, bundleCACertSecretSuffix, logr.Discard())
+
+			err := caCertBundle.Ensure(context.TODO())
+			assert.NoError(t, err)
+
+			var bundledCASecret corev1.Secret
+			err = fakeK8sClient.Get(context.TODO(), types.NamespacedName{Name: fmt.Sprintf("%s-%s", clusterName, bundleCACertSecretSuffix), Namespace: ns}, &bundledCASecret)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.expectedBundleCA, bundledCASecret.Data["ca.crt"])
+		})
+	}
+
+	err = fakeK8sClient.Delete(context.TODO(), caCert)
+	assert.NoError(t, err)
+	err = fakeK8sClient.Delete(context.TODO(), tlsCert)
+	assert.NoError(t, err)
+}
+
+func TestUpdateCACertBundle(t *testing.T) {
+	ns := "default"
+	bundleCACertSecretSuffix := "ca-bundle"
+	clusterName := "test-cluster-update"
+
+	pandaCluster := &v1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: ns,
+			UID:       "ff2770aa-c919-43f0-8b4a-30cb7cfdaf79",
```

Comment:
NIT: Can be omitted/removed

Reply:
If omitted, the unit tests fail with the error below. I haven't go chance to look into why. Any idea?
```
--- FAIL: TestUpdateCACertBundle (0.05s)
=== RUN   TestCACertBundleFailures
    ca_certificate_bundle_test.go:221:
        	Error Trace:	/Users/paulzhang/workspace/go/github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources/certmanager/ca_certificate_bundle_test.go:221
        	Error:      	Received unexpected error:
        	            	map: map[apiVersion:redpanda.vectorized.io/v1alpha1 blockOwnerDeletion:true controller:true kind:Cluster name:test-cluster-create] does not contain declared merge key: uid
        	            	Failed to generate strategic merge patch
        	            	github.com/cisco-open/k8s-objectmatcher/patch.(*PatchMaker).Calculate
        	            		/Users/paulzhang/workspace/go/pkg/mod/github.com/cisco-open/k8s-objectmatcher@v1.9.0/patch/patch.go:87
        	            	github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources.Update
        	            		/Users/paulzhang/workspace/go/github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources/resource.go:149
        	            	github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources/certmanager.(*CACertificateBundleResource).Ensure
        	            		/Users/paulzhang/workspace/go/github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources/certmanager/ca_certificate_bundle.go:84
        	            	github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources/certmanager_test.TestCACertBundleFailures
        	            		/Users/paulzhang/workspace/go/github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources/certmanager/ca_certificate_bundle_test.go:220
        	            	testing.tRunner
        	            		/usr/local/go/src/testing/testing.go:1576
        	            	runtime.goexit
        	            		/usr/local/go/src/runtime/asm_amd64.s:1598
        	Test:       	TestCACertBundleFailures

```

Reply:
You are right. It's required